packages/common/src/lib/core/jvm/utils.ts

```diff
@@ -102,18 +104,17 @@ export function getJvmPackageInfo(project: ProjectConfiguration): PackageInfo {
       /rootProject\.name\s*=\s*['"]([^"']+)['"]/
     )?.[1];
 
-    const dependencies: PackageInfo[] = [];
-    const dependencyIds =
-      buildGradle.match(
-        /\s*(api|implementation|testImplementation)\s*['"]([^"']+)['"]/
-      ) || [];
+    const dependencyIdRegEx = getDependencyIdRegEx();
+    const dependencyIds: string[] = [];
```

Comment:
Wouldn't the old logic here, have remained the same, if you simply changed the regex 
from `/\s*(api|implementation|testImplementation)\s*['"]([^"']+)['"]/`(my old value) to `/\s*(api|implementation|testImplementation)\s*\(?['"](?<id>[^"']+)['"]\)?/` (your new regex, minus the `g` global flag) ?

If yes, I would prefer keeping that part unchanged

Reply:
When capture groups appear multiple times you must iterate over the regularExpression. The previous code was the entire mach not just the value from the capture group.

If this library was targeting es2022 another way to write the cod is with matchAll & Array.form.

`const dependencyIds = Array.from(buildGradle.matchAll(dependencyIdregEx) || [], ({groups: {id}}) => id);`

Reply:
Fair enough.

Still, I see a small issue with the `do...while `construct.  When you push to `dependencyIds` you are not checking that a `match` was actually found...

A `while` construct would solve the issue

```
let match = gradleDependencyIdRegEx.exec(buildGradle);
while(match?.groups?.id){ // a while(match) would have suffice, but to be safe
	dependencyIds.push(match.groups.id);
   match = gradleDependencyIdRegEx.exec(buildGradle);
}
```

Reply:
@tinesoft 

Nice catch, forgot to check for no matches. I'm not a fan of do while either. In this case it is the best choice.

The example you provided has a couple of issues.

1. code smell: code duplication:  `match = gradleDependencyIdRegEx.exec(buildGradle);`
  a. If you remove the duplicate line then there is a lint error no setting variables inside of loop definitions.
      https://stackoverflow.com/questions/31776624/better-solution-to-regex-exec-assignment-in-while-loop
2. the new RegExp() is redundant. 
  Each time the getGradleDependencyIdRegEx is invoked it returns a new RegExp(). Think of if as a Factory.
  These 2 lines are the same because the js syntanx for /.../gi is just sugar for new RegExp("" , "");
  ```
    const regExA = /pattern/gi;
    const regExB = new RegExp("pattern", "gi");
  ``` 
  The tests demonstrates this because each test has a new instance of the Regular expression.

Reply:
Fair enough.  

In that Ok, I'll approve and merge once the build is done.
Thanks for your work ;-)